crates/sui-graphql-rpc/tests/e2e_tests.rs

```diff
@@ -416,6 +416,96 @@ mod tests {
         assert_eq!(sender_read, sender.to_string());
     }
 
+    #[tokio::test]
+    #[serial]
+    async fn test_zklogin_sig_verify() {
+        let _guard = telemetry_subscribers::TelemetryConfig::new()
+            .with_env()
+            .init();
+
+        let connection_config = ConnectionConfig::ci_integration_test_cfg();
+        let cluster =
+            sui_graphql_rpc::test_infra::cluster::start_cluster(connection_config, None).await;
+
+        // wait for epoch to be indexed, so that current epoch and JWK are populated in db.
+        let test_cluster = cluster.validator_fullnode_handle;
+        test_cluster.wait_for_epoch(Some(1)).await;
+        test_cluster.wait_for_authenticator_state_update().await;
+
+        // now query the endpoint with a valid tx data bytes and a valid signature with the correct proof for dev env.
+        let bytes = "AAABACACAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgEBAQABAAAcpgUkGBwS5nPO79YXkjMyvaRjGS57hqxzfyd2yGtejwGbB4FfBEl+LgXSLKw6oGFBCyCGjMYZFUxCocYb6ZAnFwEAAAAAAAAAIJZw7UpW1XHubORIOaY8d2+WyBNwoJ+FEAxlsa7h7JHrHKYFJBgcEuZzzu/WF5IzMr2kYxkue4asc38ndshrXo8BAAAAAAAAABAnAAAAAAAAAA==";
+        let signature = "BQNNMTczMTgwODkxMjU5NTI0MjE3MzYzNDIyNjM3MTc5MzI3MTk0Mzc3MTc4NDQyODI0MTAxODc5NTc5ODQ3NTE5Mzk5NDI4OTgyNTEyNTBNMTEzNzM5NjY2NDU0NjkxMjI1ODIwNzQwODIyOTU5ODUzODgyNTg4NDA2ODE2MTgyNjg1OTM5NzY2OTczMjU4OTIyODA5MTU2ODEyMDcBMQMCTDU5Mzk4NzExNDczNDg4MzQ5OTczNjE3MjAxMjIyMzg5ODAxNzcxNTIzMDMyNzQzMTEwNDcyNDk5MDU5NDIzODQ5MTU3Njg2OTA4OTVMNDUzMzU2ODI3MTEzNDc4NTI3ODczMTIzNDU3MDM2MTQ4MjY1MTk5Njc0MDc5MTg4ODI4NTg2NDk2Njg4NDAzMjcxNzA0OTgxMTcwOAJNMTA1NjQzODcyODUwNzE1NTU0Njk3NTM5OTA2NjE0MTA4NDAxMTg2MzU5MjU0NjY1OTcwMzcwMTgwNTg3NzAwNDEzNDc1MTg0NjEzNjhNMTI1OTczMjM1NDcyNzc1NzkxNDQ2OTg0OTYzNzIyNDI2MTUzNjgwODU4MDEzMTMzNDMxNTU3MzU1MTEzMzAwMDM4ODQ3Njc5NTc4NTQCATEBMANNMTU3OTE1ODk0NzI1NTY4MjYyNjMyMzE2NDQ3Mjg4NzMzMzc2MjkwMTUyNjk5ODQ2OTk0MDQwNzM2MjM2MDMzNTI1Mzc2Nzg4MTMxNzFMNDU0Nzg2NjQ5OTI0ODg4MTQ0OTY3NjE2MTE1ODAyNDc0ODA2MDQ4NTM3MzI1MDAyOTQyMzkwNDExMzAxNzQyMjUzOTAzNzE2MjUyNwExMXdpYVhOeklqb2lhSFIwY0hNNkx5OXBaQzUwZDJsMFkyZ3VkSFl2YjJGMWRHZ3lJaXcCMmV5SmhiR2NpT2lKU1V6STFOaUlzSW5SNWNDSTZJa3BYVkNJc0ltdHBaQ0k2SWpFaWZRTTIwNzk0Nzg4NTU5NjIwNjY5NTk2MjA2NDU3MDIyOTY2MTc2OTg2Njg4NzI3ODc2MTI4MjIzNjI4MTEzOTE2MzgwOTI3NTAyNzM3OTExCgAAAAAAAABhAG6Bf8BLuaIEgvF8Lx2jVoRWKKRIlaLlEJxgvqwq5nDX+rvzJxYAUFd7KeQBd9upNx+CHpmINkfgj26jcHbbqAy5xu4WMO8+cRFEpkjbBruyKE9ydM++5T/87lA8waSSAA==";
+        let intent_scope = 0;
+        let author = "0x1ca60524181c12e673ceefd617923332bda463192e7b86ac737f2776c86b5e8f";
```

Comment:
nit: unless we can guarantee that the data here will never change, i think we should build these params programmatically instead. this is otherwise brittle to framework changes and so on

Reply:
On the other hand, if it is possible to guarantee that these bytes remain stable, it would be better to make this a transactional test.

Reply:
unfortunately generating a zkproof is interactive, so we will have to hardcode something. i can make this to construct a transaction and use the fixed ephemeral key to produce a signature, but the zklogin proof has to be hardcoded but keep the transaction bytes itself flexible, does this work or is this what you meant by a transactional test?

Reply:
By transactional test, I meant a test like the ones in this folder:

https://github.com/MystenLabs/sui/tree/main/crates/sui-graphql-e2e-tests/tests

They are a kind of snapshot test that uses Move code to set-up some chain state and then reads that state using GraphQL.  The main motivation for switching over to that is that it makes it a little easier to follow the flow of what's supposed to happen on-chain and then what the result is, but in your case, it may not be so straightforward, so no need to work too hard to see if this is possible!

Reply:
i see! happy to follow up if this test causes issue in the future, but atm the value here should be finalized and stable for testing purpose.